internal/services/mobilenetwork/mobile_network_attached_data_network_data_source.go

```diff
@@ -0,0 +1,225 @@
+package mobilenetwork
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/mobilenetwork/2022-11-01/attacheddatanetwork"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/mobilenetwork/2022-11-01/packetcoredataplane"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type AttachedDataNetworkDataSource struct{}
+
+var _ sdk.DataSource = AttachedDataNetworkDataSource{}
+
+func (r AttachedDataNetworkDataSource) ResourceType() string {
+	return "azurerm_mobile_network_attached_data_network"
+}
+
+func (r AttachedDataNetworkDataSource) ModelObject() interface{} {
+	return &AttachedDataNetworkModel{}
+}
+
+func (r AttachedDataNetworkDataSource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return attacheddatanetwork.ValidateAttachedDataNetworkID
+}
+
+func (r AttachedDataNetworkDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"mobile_network_data_network_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"mobile_network_packet_core_data_plane_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
```

Comment:
`ForceNew` should not be set in data source, see: https://github.com/bflad/tfproviderlint/tree/main/passes/S024